compiler/testData/diagnostics/tests/smartCasts/castchecks/basicOff.fir.kt

```diff
@@ -14,15 +14,15 @@ fun g(a: SomeClass?) {
         a.hashCode()
         a.foo
         (a as? SomeSubClass)<!UNSAFE_CALL!>.<!>foo
-        (a as SomeSubClass).foo
+        (a <!USELESS_CAST!>as SomeSubClass<!>).foo
     }
     val b = (a as? SomeSubClass)?.foo
     if (b != null) {
         // 'a' can be cast to SomeSubClass
         a.hashCode()
         a.foo
         (a as? SomeSubClass)<!UNSAFE_CALL!>.<!>foo
-        (a as SomeSubClass).foo
+        (a <!USELESS_CAST!>as SomeSubClass<!>).foo
```

Comment:
Note that this test is same as `basicOn` below, but without `SafeCastCheckBoundSmartCasts`, which disables collection of nullability info. As we all know(?), nullability is not available for `== null`, but available for `!= null` in FIR DFA. That's why, with or without that flag, we're seeing those variables as non-null `SomeSubClass`. That flag is on since 1.2, so seeing the same result with `basicOn` seems okay.